src/main/java/io/gravitee/policy/jwt/revocation/RevocationChecker.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.policy.jwt.revocation;
+
+import com.nimbusds.jwt.JWTClaimsSet;
+import io.gravitee.policy.jwt.configuration.RevocationCheckConfiguration;
+import java.text.ParseException;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class RevocationChecker {
+
+    private final RevocationCheckConfiguration configuration;
+    private final RevocationCache revocationCache;
+
+    public RevocationChecker(RevocationCheckConfiguration configuration, RevocationCache revocationCache) {
+        this.configuration = configuration;
+        this.revocationCache = revocationCache;
+    }
+
+    public boolean isRevoked(JWTClaimsSet jwtClaimsSet) {
+        try {
+            if (revocationCache == null) {
+                return false;
+            }
+
+            String revocationClaim = configuration.getRevocationClaim();
+            if (revocationClaim == null || revocationClaim.isEmpty()) {
+                log.warn("No revocation claim defined in policy configuration, skipping revocation check");
+                return false;
+            }
+
+            String claimValue = jwtClaimsSet.getStringClaim(configuration.getRevocationClaim());
+            if (claimValue == null) {
+                log.warn("Claim {} not found in token, skipping revocation check", revocationClaim);
```

Comment:
I remember that this behavior is documented in the README, so it sounds like a normal case. I will leave it to debug to avoid flooding the log. Also, I would not log token claims as it is a potential security data disclosure.